packages/loopring_v3/circuit/Utils/Data.h

```diff
@@ -222,353 +340,371 @@ class Ring
     ethsnarks::FieldT fillS_B;
 };
 
-static void from_json(const json& j, Ring& ring)
+static void from_json(const json& j, SpotTrade& spotTrade)
 {
-    ring.orderA = j.at("orderA").get<Order>();
-    ring.orderB = j.at("orderB").get<Order>();
-    ring.fillS_A = ethsnarks::FieldT(j["fFillS_A"]);
-    ring.fillS_B = ethsnarks::FieldT(j["fFillS_B"]);
+    spotTrade.orderA = j.at("orderA").get<Order>();
+    spotTrade.orderB = j.at("orderB").get<Order>();
+    spotTrade.fillS_A = ethsnarks::FieldT(j["fFillS_A"]);
+    spotTrade.fillS_B = ethsnarks::FieldT(j["fFillS_B"]);
 }
 
-class RingSettlement
+class Deposit
 {
 public:
-    Ring ring;
-
-    ethsnarks::FieldT accountsMerkleRoot;
-
-    TradeHistoryUpdate tradeHistoryUpdate_A;
-    TradeHistoryUpdate tradeHistoryUpdate_B;
-
-    BalanceUpdate balanceUpdateS_A;
-    BalanceUpdate balanceUpdateB_A;
-    AccountUpdate accountUpdate_A;
-
-    BalanceUpdate balanceUpdateS_B;
-    BalanceUpdate balanceUpdateB_B;
-    AccountUpdate accountUpdate_B;
-
-    BalanceUpdate balanceUpdateA_P;
-    BalanceUpdate balanceUpdateB_P;
-
-    BalanceUpdate balanceUpdateA_O;
-    BalanceUpdate balanceUpdateB_O;
+    ethsnarks::FieldT owner;
+    ethsnarks::FieldT accountID;
+    ethsnarks::FieldT tokenID;
+    ethsnarks::FieldT amount;
+    ethsnarks::FieldT index;
 };
 
-static void from_json(const json& j, RingSettlement& ringSettlement)
+static void from_json(const json& j, Deposit& deposit)
 {
-    ringSettlement.ring = j.at("ring").get<Ring>();
-
-    ringSettlement.accountsMerkleRoot = ethsnarks::FieldT(j.at("accountsMerkleRoot").get<std::string>().c_str());
-
-    ringSettlement.tradeHistoryUpdate_A = j.at("tradeHistoryUpdate_A").get<TradeHistoryUpdate>();
-    ringSettlement.tradeHistoryUpdate_B = j.at("tradeHistoryUpdate_B").get<TradeHistoryUpdate>();
-
-    ringSettlement.balanceUpdateS_A = j.at("balanceUpdateS_A").get<BalanceUpdate>();
-    ringSettlement.balanceUpdateB_A = j.at("balanceUpdateB_A").get<BalanceUpdate>();
-    ringSettlement.accountUpdate_A = j.at("accountUpdate_A").get<AccountUpdate>();
-
-    ringSettlement.balanceUpdateS_B = j.at("balanceUpdateS_B").get<BalanceUpdate>();
-    ringSettlement.balanceUpdateB_B = j.at("balanceUpdateB_B").get<BalanceUpdate>();
-    ringSettlement.accountUpdate_B = j.at("accountUpdate_B").get<AccountUpdate>();
-
-    ringSettlement.balanceUpdateA_P = j.at("balanceUpdateA_P").get<BalanceUpdate>();
-    ringSettlement.balanceUpdateB_P = j.at("balanceUpdateB_P").get<BalanceUpdate>();
-
-    ringSettlement.balanceUpdateA_O = j.at("balanceUpdateA_O").get<BalanceUpdate>();
-    ringSettlement.balanceUpdateB_O = j.at("balanceUpdateB_O").get<BalanceUpdate>();
+    deposit.owner = ethsnarks::FieldT(j.at("owner").get<std::string>().c_str());
+    deposit.accountID = ethsnarks::FieldT(j.at("accountID"));
+    deposit.tokenID = ethsnarks::FieldT(j.at("tokenID"));
```

Comment:
What's the difference between
`ethsnarks::FieldT(j.at("tokenID"));` and
`ethsnarks::FieldT(j.at("tokenID").get<std::string>().c_str());`

Reply:
I actually want to standardize this to a single format and just use strings everywhere, but I haven't gotten around to that yet.

`ethsnarks::FieldT(j.at("tokenID"));` reads a number from the json and puts it in a field element (well, it's still a string but not a json string), and `ethsnarks::FieldT(j.at("tokenID").get<std::string>().c_str());` reads a number in a string and puts in a field element. The string method is used for data > 32bit because if I remember correctly otherwise I can't read them correctly here because of library limitations.